plugins/blockvolume/blockprovider/gluster-virtblock/gluster_virtblock.go

```diff
@@ -108,53 +105,53 @@ func (g *GlusterVirtBlk) CreateBlockVolume(name string, size uint64, hostVolume
 		}
 	}
 
-	if err = clusterLocks.Lock(hostVolume); err != nil {
-		logger.WithError(err).Error("error in acquiring cluster lock")
-		return nil, err
+	return &BlockVolume{
+		hostVolume: hostVolume,
+		name:       name,
+		size:       size,
+	}, nil
+}
+
+// GetAndDeleteBlockVolume deletes a gluster block volume of give name
+func (g *GlusterVirtBlk) GetAndDeleteBlockVolume(name string, options ...blockprovider.BlockVolOption) (blockprovider.BlockVolume, error) {
```

Comment:
same here 

this function and GetAndDeleteBlockVolume looks similar, any major difference, can we avoid code duplication?